syntax = "proto3";

// enum PhoneType {
//     Mobile = 0;
//     Home = 1;
//     Work = 2;
// }



enum MsgType{
    BaseMsgType = 0;
    Test1MsgType = 1;
    Test2MsgType = 2;
}

message BaseMsg {
    MsgType msgType = 1;
}

message Test1Msg {
    MsgType msgType = 1;
    string name = 2;
    string text = 3;
}

message Test2Msg {
    MsgType msgType = 1;
    string name = 2;
    string pwd = 3;
}